first-party/jib-layer-filter-extension-maven/src/test/java/com/google/cloud/tools/jib/maven/extension/layerfilter/JibLayerFilterExtensionTest.java

```diff
@@ -291,4 +326,115 @@ public void testExtendContainerBuildPlan_complex() throws JibPluginExtensionExce
     assertEquals(Arrays.asList("/gamma/Charlie"), layerToExtractionPaths(newLayer6));
   }
+
+  @Test
+  public void testExtendContainerBuildPlan_createParentLayers_noProjectDependenciesResolver() {
+    ContainerBuildPlan buildPlan = ContainerBuildPlan.builder().build();
+    Configuration cfg = new Configuration();
+    cfg.setCreateParentLayers(true);
+    try {
+      new JibLayerFilterExtension()
+          .extendContainerBuildPlan(buildPlan, null, Optional.of(cfg), mavenData, logger);
+      fail();
+    } catch (JibPluginExtensionException ex) {
+      assertEquals(JibLayerFilterExtension.class, ex.getExtensionClass());
+      assertEquals(
+          "Try to get parent dependencies, but ProjectDependenciesResolver is null. Please use a more recent jib plugin version to fix this.",
+          ex.getMessage());
+    }
+  }
+
+  @Test
+  public void testExtendContainerBuildPlan_createParentLayers_noParent()
+      throws JibPluginExtensionException {
+    ContainerBuildPlan buildPlan = ContainerBuildPlan.builder().build();
+    Configuration cfg = new Configuration();
+    cfg.setCreateParentLayers(true);
+
+    when(mavenProject.getParent()).thenReturn(null);
+
+    try {
+      JibLayerFilterExtension extension = new JibLayerFilterExtension();
+      extension.setDependencyResolver(projectDependenciesResolver);
+
+      extension.extendContainerBuildPlan(buildPlan, null, Optional.of(cfg), mavenData, logger);
+      fail();
+    } catch (JibPluginExtensionException ex) {
+      assertEquals(JibLayerFilterExtension.class, ex.getExtensionClass());
+      assertEquals("Try to get parent dependencies, but project has no parent.", ex.getMessage());
+    }
+  }
+
+  @Test
+  public void testExtendContainerBuildPlan_createParentLayers_noParentDependencies()
+      throws JibPluginExtensionException {
+    FileEntriesLayer layer1 = buildLayer("", Arrays.asList("/app/libs/foo-1.0.0.jar"));
+    ContainerBuildPlan buildPlan = ContainerBuildPlan.builder().addLayer(layer1).build();
+    Configuration cfg = new Configuration();
+    cfg.setCreateParentLayers(true);
+
+    when(dependencyResolutionResult.getDependencies()).thenReturn(Collections.emptyList());
+
+    JibLayerFilterExtension extension = new JibLayerFilterExtension();
+    extension.setDependencyResolver(projectDependenciesResolver);
+
+    ContainerBuildPlan newPlan =
+        extension.extendContainerBuildPlan(buildPlan, null, Optional.of(cfg), mavenData, logger);
+
+    assertEquals(1, newPlan.getLayers().size());
+    FileEntriesLayer newLayer1 = (FileEntriesLayer) newPlan.getLayers().get(0);
+    assertEquals("", newLayer1.getName());
+    assertEquals(layer1.getEntries(), newLayer1.getEntries());
+  }
+
+  @Test
+  public void testExtendContainerBuildPlan_createParentLayers_withParentDependencies()
+      throws JibPluginExtensionException {
+    FileEntriesLayer layer1 =
+        buildLayer(
+            "layer1",
+            asList(
```

Comment:
no static imports unless test gadgets
```suggestion
        Arrays.asList(
```
I see there are many occurrences of this below.

Reply:
ok